src/serde/mod.rs

```diff
@@ -12,12 +12,70 @@ mod tools;
 mod utils;
 pub mod write_atom;
 
-#[cfg(test)]
-mod test;
-
 pub use de::node_from_bytes;
 pub use de_br::node_from_bytes_backrefs;
 pub use de_tree::{parse_triples, ParsedTriple};
 pub use ser::node_to_bytes;
 pub use ser_br::node_to_bytes_backrefs;
 pub use tools::{serialized_length_from_bytes, tree_hash_from_stream};
+
+#[cfg(test)]
+mod tests {
+    use crate::Allocator;
+
+    use super::*;
+
+    use hex::FromHex;
+
+    fn check(serialized_backrefs_hex: &str) {
+        // Serialized with br => obj => serialized no br =(allow_br)=> obj => serialized w br.
+
+        // Serialized object, with back-refs.
+        let serialized_backrefs_1 = <Vec<u8>>::from_hex(serialized_backrefs_hex).unwrap();
+
+        // Turn into serialized object with no back-refs.
+        let mut allocator = Allocator::new();
+        let object = node_from_bytes_backrefs(&mut allocator, &serialized_backrefs_1).unwrap();
+
+        let serialized_1 = node_to_bytes(&allocator, object).unwrap();
+
+        // Deserialize using `node_from_bytes_backrefs` (even though there are no backrefs)
+        // and reserialized without back-refs.
+        let mut allocator = Allocator::new();
+        let object = node_from_bytes_backrefs(&mut allocator, &serialized_1).unwrap();
+
+        let serialized_2 = node_to_bytes(&allocator, object).unwrap();
+
+        // Compare both reserializations (without back-refs).
+        assert_eq!(serialized_1, serialized_2);
+
+        // Now reserialize with back-refs.
+        let mut allocator = Allocator::new();
+        let obj = node_from_bytes(&mut allocator, &serialized_1).unwrap();
+
+        let serialized_backrefs_2 = node_to_bytes_backrefs(&allocator, obj).unwrap();
+
+        // And compare to original.
+        assert_eq!(serialized_backrefs_1, serialized_backrefs_2);
+    }
+
+    #[test]
+    fn test_round_trip() {
```

Comment:
did you add this test? where does it come from?
(I think it's risky to mix moving functions with any other changes in a single commit)

Reply:
It was just moved. Next time I'll do separate commits.